pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -121,36 +121,31 @@ let
       src = fetchurl { inherit urls sha512; };
 
       passthru = {
-        inherit pname tlType version;
-      } // lib.optionalAttrs (sha512 != "") { inherit src; };
+        inherit pname src tlType version;
```

Comment:
```suggestion
        inherit tlType;
```

Reply:
Will removing `src` have side effects with `find-tarballs`? I.e. the original tarballs won't be mirrored? Just to know if this should be a separate commit.

Reply:
That is a good question. find-tarballs will ignore `fetchurl` that has non-empty `postFetch`
https://github.com/NixOS/nixpkgs/blob/be976cdda9fcb41402be7ba1d09166e094e5e0f5/maintainers/scripts/find-tarballs.nix#L19-L23
At the same time, the hashed mirrors are queried with the hash that was supplied to the `fetchurl`, which in this case is not the hash of the original tarball.

Basically we are going to lose the tarballs.nixos.org backup if we go with the `fixedHash != null` condition.

Reply:
Ah, right. Uhm, what if we do `fetchurl { ... } // { inherit src; }`? Will `find-tarballs.nix` catch `src` as "immediate dependency"?

Reply:
I have figured out how to run `find-tarballs.nix`. I might be right on this one:
```bash
nix-instantiate --json --strict --eval maintainers/scripts/find-tarballs.nix --arg expr '(import ./. {}).texlive.combined.scheme-full' | jq
```
shows the original texlive tarballs. But they disappear if I remove `src` from `passthru`.

Do we want the tarballs to be backed up? I suppose the answer is yes?

Reply:
> I have figured out how to run `find-tarballs.nix`. I might be right on this one:
> 
> ```shell
> nix-instantiate --json --strict --eval maintainers/scripts/find-tarballs.nix --arg expr '(import ./. {}).texlive.combined.scheme-full' | jq
> ```
> 
> shows the original texlive tarballs. But they disappear if I remove `src` from `passthru`.

Sure, but the `fetchurl` is not going to download them if the supplied hash is of the recursive hash of the unpacked directory.

> 
> Do we want the tarballs to be backed up?

I, personally, would prefer if they were backed up.

Reply:
Right, so `passthru.src` seems to give us the backup, but `fetchurl` won't use it.

I'd say, let's leave `passthru.src` in place so that at least the backup exists. We could add a flag similar to `useFixedHashes` to disable fetch&unpack, but keep the fixed hashes, to get some level of reproducibility.

Reply:
Well, backup without restore is not really a backup.

The only benefit of combining fetch and unpack is to save some disk space, but if we allow the unpacked NAR to be fetched from the Hydra, it should get the same effect for most of our users. Users who don't substitute from Hydra are also the only ones who can run into problems with unavailable upstream tarballs, so it would make sense to go the "long way" for those.

Reply:
Just to make sure I am following, are you saying that we should disable fetch&unpack and allow substitutes in the other derivation (essentially, revert to my first commit)?

It sounds like the only use case of fetch&unpack is for users who don't/can't substitute *and* are space constrained.

Reply:
> Just to make sure I am following, are you saying that we should disable fetch&unpack and allow substitutes in the other derivation (essentially, revert to my first commit)?

Yes, your original PR sounds better in the light of our discussion. Maybe with some docs on why it is what it is. Or we could drop the "fetch&unpack" branch altogether.

Reply:
Ok, let's start with the minimum possible change so that it's trivial to backport to 21.05. I can remove fetch&unpack (and simplify the surrounding code) in a second PR.

Reply:
I am also not sure if tarballs.nixos.org even serves tarballs for sha512 hashes, it might not.

Reply:
> I am also not sure if tarballs.nixos.org even serves tarballs for sha512 hashes, it might not.

It looks like it should

https://github.com/NixOS/nixpkgs/blob/8a362be5fee28b7081fe2c0a37a6b0c53498a4ae/maintainers/scripts/copy-tarballs.pl#L116

OTOH, are tarballs stuck at 19.09?
https://github.com/NixOS/nixos-org-configurations/blob/014895e4efe894f5dd4d026b78ab86bbecf38488/modules/tarball-mirror.nix#L14